src/classes/BDI_DataImportFLSService.cls

```diff
@@ -136,15 +140,16 @@ public class BDI_DataImportFLSService {
         }//END loop of Data Import records
     }
 
-    private Boolean isValid(SObjectField field) {
+    private Boolean isValid(SObjectField field, Boolean isCampaignMemberByPass) {
         Set<SObjectField> fieldSet = new Set<SObjectField>{
                 field
         };
         if ((accessLevels.contains(AccessLevel.VALIDATE_CREATE)
                 && !permissionsService.canCreate(fieldSet))
                 ||
                 (accessLevels.contains(AccessLevel.VALIDATE_UPDATE)
-                        && !permissionsService.canUpdate(fieldSet)
+                        && (!permissionsService.canUpdate(fieldSet)
+                        && !isCampaignMemberByPass)
```

Comment:
Since we now have multiple exception cases when checking update permissions, what do you think about adding a new private method called something like `isValidateUpdateException` that would contain the master detail check below _and_ this new check for the CampaignId field?  Instead of changing the signature of this `isValid` method to accept this new boolean, it could be changed to accept both the SObjectType and SObjectFieldDescribe.  Both could be passed to the new isValidateUpdateException method to see if either condition is met.  That way the false boolean value wouldn't need to be passed into this isValid method for all other fields.

Reply:
Hey Ken, this makes sense! Did you get my message on quip?